### PR TITLE
Fix create branch API

### PR DESCRIPTION
**Pull request by sotayamashita**
## Create a branch

```ruby
g = Gitlab.client(endpoint: 'https://gitlab.com/api/v4', private_token: 'xxxx')

# Create a branch
g.create_branch(5221359, "newbranch", "master")
=> #<Gitlab::ObjectifiedHash:70145150190240 {hash: {"name"=>"newbranch", ~ 
```
- https://docs.gitlab.com/ce/api/branches.html#create-repository-branch

Thanks in advance :bow:

### CODE REVIEW

File: lib/gitlab/client/branches.rb [hunk 1]
```diff
@@ -47,7 +47,7 @@ def branch(project, branch)
     # @option options [Boolean] :developers_can_merge True to allow developers to merge into the branch (default = false)
     # @return [Gitlab::ObjectifiedHash] Details about the branch
     def protect_branch(project, branch, options = {})
-      post("/projects/#{url_encode project}/protected_branches", body: {name: branch}.merge(options))
+      put("/projects/#{url_encode project}/repository/branches/#{url_encode branch}/protect", query: options)
```
Comment (NARKOZ) [inline lib/gitlab/client/branches.rb]: Docs are saying that this API endpoint is deprecated in favor of `POST /projects/:id/protected_branches`. https://docs.gitlab.com/ce/api/branches.html#protect-repository-branch

File: lib/gitlab/client/branches.rb [hunk 2]
```diff
@@ -61,22 +61,22 @@ def protect_branch(project, branch, options = {})
     # @param  [String] branch The name of the branch.
     # @return [Gitlab::ObjectifiedHash] Details about the branch
     def unprotect_branch(project, branch)
-      delete("/projects/#{url_encode project}/protected_branches/#{url_encode branch}")
+      put("/projects/#{url_encode project}/repository/branches/#{url_encode branch}/unprotect")
```
Comment (NARKOZ) [inline lib/gitlab/client/branches.rb]: Docs are saying that this API endpoint is deprecated in favor of `DELETE /projects/:id/protected_branches/:name`. https://docs.gitlab.com/ce/api/branches.html#unprotect-repository-branch

**Comment by sotayamashita** [comment]
@NARKOZ Could you review it again?

**Comment by NARKOZ** [comment]
Thank you 👍

**Pull request merged**